Sources/MessagingPush/MessagingPushImplementation.swift

```diff
@@ -49,12 +57,17 @@ internal class MessagingPushImplementation: MessagingPushInstance {
      Delete the currently registered device token
      */
     public func deleteDeviceToken() {
+        logger.info("deleting device token request made")
+
         guard let existingDeviceToken = globalDataStore.pushDeviceToken else {
+            logger.info("no device token exists so ignoring request to delete")
             return // no device token to delete, ignore request
         }
-        globalDataStore.pushDeviceToken = nil
+        // Do not delete push token from device storage. The token is valid
+        // once given to SDK. We need it for future profile identifications.
```

Comment:
this is a bug that I found after implementing logging and I did some QA testing of the queue myself.